### PR TITLE
Update the kubeadm documentation to v1.5

### PR DESCRIPTION
**Pull request by luxas**
@kubernetes/sig-cluster-lifecycle PTAL

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1968)

### CODE REVIEW

File: docs/admin/kubeadm.md [hunk 1]
```diff
@@ -168,20 +168,71 @@ necessary.
 By default, when `kubeadm init` runs, a token is generated and revealed in the output.
 That's the token you should use here.
 
+
+## Using kubeadm with a config file
+
+It's possible to configure kubeadm with a config file instead of command line flags, and the more advanced modes of kubeadm will only be
```
Comment (dgoodwin) [inline docs/admin/kubeadm.md]: Suggest: "and some more advanced features may only be available as configuration file options", and replacing "config" with "configuration".

File: docs/admin/kubeadm.md [hunk 2]
```diff
@@ -168,20 +168,71 @@ necessary.
...
+available from as configuration file options. Here is a demonstration of what the config file looks like.
```
Comment (dgoodwin) [inline docs/admin/kubeadm.md]: In any case need to drop the "from" here.

File: docs/admin/kubeadm.md [hunk 3]
```diff
@@ -168,20 +168,71 @@ necessary.
...
+
+Keep in mind that this is only the alpha version and this API syntax will _definitely_ change.
```
Comment (dgoodwin) [inline docs/admin/kubeadm.md]: Suggest: 

WARNING: kubeadm is in alpha and the configuration API syntax will likely change before GA.

File: docs/admin/kubeadm.md (hunk 2 above)
Comment (dgoodwin) [inline docs/admin/kubeadm.md]: I would drop the "Here is a demonstration" and instead do a:

### Sample Configuration

File: docs/admin/kubeadm.md [hunk 4]
```diff
@@ -168,20 +168,71 @@ necessary.
...
 in the basic `kubeadm` tutorials, you can parallelize the token distribution for
 easier automation. To implement this automation, you must know the IP address
 that the master will have after it is started.
 
-1.  Generate a token.  This token must have the form  `<6 character string>.<16
-character string>`
+1.  Generate a token.  This token must have the form  `<6 character string>.<16 character string>`.
 
-    Here is a simple python one-liner for this:
+    Here is how you can do it easily with kubeadm:
```
Comment (dgoodwin) [inline docs/admin/kubeadm.md]: Suggest: "Kubeadm can pre-generate a token for you:"

File: docs/getting-started-guides/kubeadm.md [hunk 5]
```diff
@@ -277,53 +296,67 @@ See the [list of add-ons](/docs/admin/addons/) to explore other add-ons, includi
...
 Replace `ARCH=amd64` with `ARCH=arm` or `ARCH=arm64` depending on the platform you're running on.
 Note that the Raspberry Pi 3 is in ARM 32-bit mode, so for RPi 3 you should set `ARCH` to `arm`, not `arm64`.
 
-## Limitations
+## Cloudprovider integrations (experimental)
 
-Please note: `kubeadm` is a work in progress and these limitations will be addressed in due course.
-Also you can take a look at the troubleshooting section in the [reference document](/docs/admin/kubeadm/#troubleshooting)
+We have got a lot of requests of how to enable a specific cloudprovider. This is possible, but currently requires manual configuration, 
+and therefore not supported. As a first step if you want to do it, edit the `kubeadm` dropin for the `kubelet` service
```
Comment (dgoodwin) [inline docs/getting-started-guides/kubeadm.md]: Suggest: "Enabling specific cloud providers is a common request, this currently requires manual configuration and is therefore not yet supported. If you wish to do so, edit the..."

File: docs/getting-started-guides/kubeadm.md [hunk 6]
```diff
@@ -277,53 +296,67 @@ See the [list of add-ons](/docs/admin/addons/) to explore other add-ons, includi
...
-   To set up kubeadm with CloudProvider integrations (it's experimental, but try), refer to the [kubeadm reference](/docs/admin/kubeadm/) document.
+Specify the `--cloud-provider` flag to kubelet and set it to the cloud of your choice. If your cloudprovider requires a configuration
+file, create the file `/etc/kubernetes/cloud-config` on every node and set the values your cloud requires. Also append
+`--cloud-config=/etc/kubernetes/cloud-config` to the kubelet arguments.
 
-   Workaround: use the [NodePort feature of services](/docs/user-guide/services/#type-nodeport) for exposing applications to the internet.
+Lastly, run `kubeadm init --cloud-provider=xxx` to bootstrap your cluster with cloud provider features.
+
+The reason we're not fully supporting this workflow is that it's very manual process, and it doesn't fit into the rest of the kubeadm
+flow really. But we're working on making it super-easy to spin up a cluster with cloudprovider power-ups. See
```
Comment (dgoodwin) [inline docs/getting-started-guides/kubeadm.md]: Suggest: This workflow is not yet fully supported, however we hope to make it extremely easy to spin up clusters with cloud providers in the future. (See this proposal for more information) The Kubelet Dynamic Settings feature may also help to fully automate this process in the future.

**Comment by luxas** [comment]
Looks good now? Thanks for the comments
I'd want to get this in so the release process is fully finished.

**Comment by dgoodwin** [comment]
LGTM

**Comment by luxas** [comment]
Great!

**Comment by carllhw** [comment]
Good job!

**Pull request merged**